cmd/mock.go

```diff
@@ -0,0 +1,308 @@
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+
+*/
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"path"
+	"path/filepath"
+	"strconv"
+	"syscall"
+
+	"github.com/docker/docker/client"
+	"github.com/fsnotify/fsnotify"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/kubeshop/kusk/internal/config"
+	"github.com/kubeshop/kusk/internal/mocking"
+	"github.com/kubeshop/testkube/pkg/ui"
+	"github.com/spf13/cobra"
+
+	"github.com/kubeshop/kusk-gateway/pkg/spec"
+	mockingServer "github.com/kubeshop/kusk/internal/mocking/server"
+)
+
+// mockCmd represents the mock command
+var mockCmd = &cobra.Command{
+	Use:   "mock",
+	Short: "Spin up a local mocking server serving your API",
+	Long: `Spin up a local mocking server that generates responses from your content schema or returns your defined examples.
+Schema example:
+
+content:
+ application/json:
+  schema:
+   type: object
+   properties:
+    title:
+     type: string
+     description: Description of what to do
+    completed:
+     type: boolean
+    order:
+     type: integer
+     format: int32
+    url:
+     type: string
+     format: uri
+   required:
+    - title
+    - completed
+    - order
+    - url
+
+The mock server will return a response like the following that matches the schema above:
+{
+ "completed": false,
+ "order": 1957493166,
+ "title": "Inventore ut.",
+ "url": "http://langosh.name/andreanne.parker"
+}
+
+Example with example responses:
+
+application/xml:
+ example:
+  title: "Mocked XML title"
+  completed: true
+  order: 13
+  url: "http://mockedURL.com"
+
+The mock server will return this exact response as its specified in an example:
+<doc>
+ <completed>true</completed>
+ <order>13</order>
+ <title>Mocked XML title</title>
+ <url>http://mockedURL.com</url>
+</doc>
+`,
+	Example: "kusk mock -i path-to-openapi-file.yaml",
+	Run: func(cmd *cobra.Command, args []string) {
+		homeDir, err := os.UserHomeDir()
```

Comment:
small nit: Instead of writing config to homeDir maybe we can use tmp?

I imagine users dont need to care about this config?

Reply:
home dir isn't too bad if they want to reuse it and/or set some global variables there
I wouldn't touch it any further